### ISSUE TITLE
Import errors when running `python setup.py test`

### ISSUE THREAD

**Issue by serteckian** [htm-community/htm.core#907]
Hello, 

I am able to build  `python setup.py install` in a fresh installation with no issues. However, when I run the tests with `python setup.py test`, I get many import errors (see below). 

In addition, just for testing, I have also been able to built C++ with `make -j4 install` with no issues. 

When I try to import the packages from the python shell, it seems that the only available package is `htm.bindings.sys`.

I am running python 3.8 (see output of `pip freeze` below).

```
[----------] Global test environment tear-down
[==========] 389 tests from 44 test cases ran. (36829 ms total)
[  PASSED  ] 389 tests.

  YOU HAVE 2 DISABLED TESTS

============================================================= test session starts =============================================================
platform linux -- Python 3.8.0, pytest-4.6.5, py-1.10.0, pluggy-0.13.1 -- /home/love/PycharmProjects/htm/htm.core/venv3.8/bin/python
cachedir: .pytest_cache
rootdir: /home/love/PycharmProjects/htm/htm.core, inifile: setup.cfg
collected 5 items / 15 errors                                                                                                                 

=================================================================== ERRORS ====================================================================
___________________________________________ ERROR collecting bindings/py/tests/nupic_random_test.py ___________________________________________
ImportError while importing test module '/home/user/htm/htm.core/bindings/py/tests/nupic_random_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
nupic_random_test.py:28: in <module>
    from htm.bindings.math import Random
E   ModuleNotFoundError: No module named 'htm.bindings.math'
___________________________________________ ERROR collecting bindings/py/tests/sparse_link_test.py ____________________________________________
ImportError while importing test module '/home/user/htm/htm.core/bindings/py/tests/sparse_link_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
sparse_link_test.py:23: in <module>
    import htm.bindings.engine_internal as engine
E   ModuleNotFoundError: No module named 'htm.bindings.engine_internal'
_____________________________________________ ERROR collecting bindings/py/tests/topology_test.py _____________________________________________
ImportError while importing test module '/home/user/htm/htm.core/bindings/py/tests/topology_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
topology_test.py:22: in <module>
    from htm.bindings.math import (DefaultTopology,
E   ModuleNotFoundError: No module named 'htm.bindings.math'
______________________________________ ERROR collecting bindings/py/tests/algorithms/connections_test.py ______________________________________
ImportError while importing test module '/home/user/htm/htm.core/bindings/py/tests/algorithms/connections_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
algorithms/connections_test.py:22: in <module>
    from htm.bindings.sdr import SDR
E   ModuleNotFoundError: No module named 'htm.bindings.sdr'
____________________________________ ERROR collecting bindings/py/tests/algorithms/sdr_classifier_test.py _____________________________________
ImportError while importing test module '/home/user/htm/htm.core/bindings/py/tests/algorithms/sdr_classifier_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
algorithms/sdr_classifier_test.py:29: in <module>
    from htm.bindings.sdr import SDR
E   ModuleNotFoundError: No module named 'htm.bindings.sdr'
____________________________________ ERROR collecting bindings/py/tests/algorithms/spatial_pooler_test.py _____________________________________
ImportError while importing test module '/home/user/htm/htm.core/bindings/py/tests/algorithms/spatial_pooler_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
algorithms/spatial_pooler_test.py:24: in <module>
    from htm.bindings.sdr import SDR
E   ModuleNotFoundError: No module named 'htm.bindings.sdr'
____________________________________ ERROR collecting bindings/py/tests/algorithms/temporal_memory_test.py ____________________________________
ImportError while importing test module '/home/user/htm/htm.core/bindings/py/tests/algorithms/temporal_memory_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
algorithms/temporal_memory_test.py:24: in <module>
    from htm.bindings.sdr import SDR
E   ModuleNotFoundError: No module named 'htm.bindings.sdr'
______________________________________ ERROR collecting bindings/py/tests/encoders/date_encoder_test.py _______________________________________
ImportError while importing test module '/home/user/htm/htm.core/bindings/py/tests/encoders/date_encoder_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
encoders/date_encoder_test.py:24: in <module>
    from htm.bindings.encoders import DateEncoder, DateEncoderParameters
E   ModuleNotFoundError: No module named 'htm.bindings.encoders'
__________________________________________ ERROR collecting bindings/py/tests/encoders/rdse_test.py ___________________________________________
ImportError while importing test module '/home/user/htm/htm.core/bindings/py/tests/encoders/rdse_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
encoders/rdse_test.py:24: in <module>
    from htm.bindings.sdr import SDR, Metrics
E   ModuleNotFoundError: No module named 'htm.bindings.sdr'
_____________________________________ ERROR collecting bindings/py/tests/encoders/scalar_encoder_test.py ______________________________________
ImportError while importing test module '/home/user/htm/htm.core/bindings/py/tests/encoders/scalar_encoder_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
encoders/scalar_encoder_test.py:26: in <module>
    from htm.encoders.scalar_encoder import ScalarEncoder, ScalarEncoderParameters
../../../build/Release/distr/src/htm/encoders/__init__.py:1: in <module>
    from htm.bindings.encoders import __doc__
E   ModuleNotFoundError: No module named 'htm.bindings.encoders'
________________________________ ERROR collecting bindings/py/tests/encoders/simhash_document_encoder_test.py _________________________________
ImportError while importing test module '/home/user/htm/htm.core/bindings/py/tests/encoders/simhash_document_encoder_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
encoders/simhash_document_encoder_test.py:28: in <module>
    from htm.bindings.encoders import SimHashDocumentEncoder, \
E   ModuleNotFoundError: No module named 'htm.bindings.encoders'
_________________________________________ ERROR collecting bindings/py/tests/regions/network_test.py __________________________________________
ImportError while importing test module '/home/user/htm/htm.core/bindings/py/tests/regions/network_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
regions/network_test.py:28: in <module>
    from htm.bindings.sdr import SDR
E   ModuleNotFoundError: No module named 'htm.bindings.sdr'
________________________________________ ERROR collecting bindings/py/tests/regions/sp_region_test.py _________________________________________
ImportError while importing test module '/home/user/htm/htm.core/bindings/py/tests/regions/sp_region_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
regions/sp_region_test.py:22: in <module>
    from htm.bindings.engine_internal import Network
E   ModuleNotFoundError: No module named 'htm.bindings.engine_internal'
___________________________________________ ERROR collecting bindings/py/tests/sdr/Metrics_test.py ____________________________________________
ImportError while importing test module '/home/user/htm/htm.core/bindings/py/tests/sdr/Metrics_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
sdr/Metrics_test.py:25: in <module>
    from htm.bindings import sdr
E   ImportError: cannot import name 'sdr' from 'htm.bindings' (/home/user/htm/htm.core/build/Release/distr/src/htm/bindings/__init__.py)
_____________________________________________ ERROR collecting bindings/py/tests/sdr/SDR_test.py ______________________________________________
ImportError while importing test module '/home/user/htm/htm.core/bindings/py/tests/sdr/SDR_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
sdr/SDR_test.py:27: in <module>
    from htm.bindings.sdr import SDR
E   ModuleNotFoundError: No module named 'htm.bindings.sdr'
```

```
$pip freeze
atomicwrites==1.4.0
attrs==20.3.0
certifi==2020.12.5
chardet==4.0.0
cmake==3.18.4.post1
Hexy==1.4.3
htm.core==2.1.15
idna==2.10
importlib-metadata==3.3.0
mock==4.0.3
more-itertools==8.6.0
numpy==1.19.4
packaging==20.8
pluggy==0.13.1
prettytable==2.0.0
py==1.10.0
pyparsing==2.4.7
pytest==4.6.5
requests==2.25.1
six==1.15.0
urllib3==1.26.2
wcwidth==0.2.5
zipp==3.4.0

**Comment by dkeeney** [comment]
Hmmm, does not seem to be finding the bindings.
I will see if I can duplicate this problem.

**Comment by dkeeney** [comment]
When you build with CMake and make install it will build the C++ library but not the python modules.

Here is what I suggest:
1 Delete the entire 'build' directory.  This is to clear the CMake cache.
2 execute this command at a command prompt
```
   python setup.py install
```
 
Then `python setup.py test` should run.

**Comment by serteckian** [comment]
Hi David,

Thanks for the quick response. I've tried removing the entire 'build' directory, and running `python setup.py install` and the tests still fail. I tried removing 'build' and installing with a new venv on python v3.8 but failed. Then I removed 'build' again and tried with an earlier venv with python v3.7, which also failed. 

Finally, I cloned the repository in a new directory, created a new venv with python v3.7, installed and all tests passed just fine... magic! No idea what could have caused the error.

Solved. Thanks!
